File: api/src/main/java/fr/huiitre/tools/modules/riot/valorant/api/ValorantUserSkinController.java
package fr.huiitre.tools.modules.riot.valorant.api;

import fr.huiitre.tools.modules.core.common.api.RequiredRole;
import fr.huiitre.tools.modules.core.role.domain.RoleCode;
import fr.huiitre.tools.modules.riot.valorant.application.user.command.AddUserSkinCommand;
import fr.huiitre.tools.modules.riot.valorant.application.user.usecase.AddMyValorantSkinUseCase;
import fr.huiitre.tools.modules.riot.valorant.application.user.usecase.GetMyValorantUserSkinsUseCase;
import fr.huiitre.tools.modules.riot.valorant.application.user.usecase.RemoveMyValorantSkinUseCase;
import fr.huiitre.tools.modules.riot.valorant.application.skin.view.ValorantSkinView;
import org.springframework.http.HttpStatus;
import org.springframework.web.bind.annotation.*;

import java.util.List;

@RestController
@RequestMapping("/riot/valorant/my-skins")
public class ValorantUserSkinController {

    private final GetMyValorantUserSkinsUseCase getMyValorantUserSkinsUseCase;
    private final AddMyValorantSkinUseCase addMyValorantSkinUseCase;
    private final RemoveMyValorantSkinUseCase removeMyValorantSkinUseCase;

    public ValorantUserSkinController(
            GetMyValorantUserSkinsUseCase getMyValorantUserSkinsUseCase,
            AddMyValorantSkinUseCase addMyValorantSkinUseCase,
            RemoveMyValorantSkinUseCase removeMyValorantSkinUseCase) {
        this.getMyValorantUserSkinsUseCase = getMyValorantUserSkinsUseCase;
        this.addMyValorantSkinUseCase = addMyValorantSkinUseCase;
        this.removeMyValorantSkinUseCase = removeMyValorantSkinUseCase;
    }

    @RequiredRole(RoleCode.READ_ONLY)
    @GetMapping
    public List<ValorantSkinView> getMyUserSkins() {
        return getMyValorantUserSkinsUseCase.execute();
    }

    @RequiredRole(RoleCode.USER)
    @PostMapping
    @ResponseStatus(HttpStatus.CREATED)
    public ValorantSkinView addMySkin(@RequestBody AddUserSkinCommand command) {
        return addMyValorantSkinUseCase.execute(command);
    }

    @RequiredRole(RoleCode.USER)
    @DeleteMapping("/{skinId}")
    @ResponseStatus(HttpStatus.NO_CONTENT)
    public void removeMySkin(@PathVariable Long skinId) {
        removeMyValorantSkinUseCase.execute(skinId);
    }
}

File: api/Modules/Riot/Valorant/Api/ValorantUserSkinController.cs
using Microsoft.AspNetCore.Mvc;
using Tools.Api.Modules.Riot.Valorant.Application.Skin.Views;
using Tools.Api.Modules.Riot.Valorant.Application.User.Commands;
using Tools.Api.Modules.Riot.Valorant.Application.User.Usecases;

namespace Tools.Api.Modules.Riot.Valorant.Api;

// Skins possédés, par compte Valorant lié.
[ApiController]
[Route("riot/valorant/my-skins")]
public class ValorantUserSkinController : ControllerBase
{
    [HttpGet]
    public Task<List<ValorantSkinView>> GetMyUserSkins(
        [FromServices] GetMyValorantUserSkinsUseCase getMyValorantUserSkinsUseCase,
        [FromQuery] long accountId
    )
    {
        return getMyValorantUserSkinsUseCase.Execute(accountId);
    }

    [HttpPost]
    public async Task<IActionResult> AddMySkin(
        [FromServices] AddMyValorantSkinUseCase addMyValorantSkinUseCase,
        [FromBody] AddUserSkinCommand command
    )
    {
        var skin = await addMyValorantSkinUseCase.Execute(command);

        return StatusCode(StatusCodes.Status201Created, skin);
    }

    [HttpDelete("{skinId:long}")]
    public async Task<IActionResult> RemoveMySkin(
        [FromServices] RemoveMyValorantSkinUseCase removeMyValorantSkinUseCase,
        [FromRoute] long skinId,
        [FromQuery] long accountId
    )
    {
        await removeMyValorantSkinUseCase.Execute(skinId, accountId);

        return NoContent();
    }
}
